### ISSUE TITLE
HDFS Support?

### ISSUE THREAD

**Issue by andygrove** [sunchao/parquet-rs#93]
I'm curious if there are plans to support HDFS within this crate?

The Java parquet library allows parquet files to be read locally or from HDFS and in both cases it is possible to push down the projection and only retrieve the columns needed, which can make a huge difference in performance.

**Comment by sunchao** [comment]
Yes that's in the plan. I took a brief look at the hdfs crate, and think it should be relatively easy to have [`HdfsFile`](http://hyunsik.github.io/hdfs-rs/hdfs/struct.HdfsFile.html) implement the [`Read`](https://doc.rust-lang.org/std/io/trait.Read.html). With that, it seems to only require a small code change.

The projection pushdown is orthogonal to the HDFS issue I think. It is [already supported](https://github.com/sunchao/parquet-rs/blob/master/src/file/reader.rs#L62) by the current record reader.
